js/components/boundary-config-dialog.tsx

```diff
@@ -0,0 +1,143 @@
+import * as React from "react";
+import Draggable from "react-draggable";
+import Dialog from "@mui/material/Dialog";
+import DialogActions from "@mui/material/DialogActions";
+import DialogContentText from "@mui/material/DialogContentText";
+import DialogTitle from "@mui/material/DialogTitle";
+import Paper, { PaperProps } from "@mui/material/Paper";
+import { BoundaryOrientation, BoundaryType, IBoundaryInfo } from "../types";
+import CloseIcon from "../../images/rock-key/svg/close-icon.svg";
+import BoundarySvg from "../../images/boundary.svg";
+import PlateArrow from "../../images/plate-arrow.svg";
+
+import css from "../../css-modules/boundary-config-dialog.less";
+
+function PaperComponent(props: PaperProps) {
+  return (
+    <Draggable
+      handle="#draggable-dialog-title"
+      cancel={'[class*="MuiDialogContent-root"]'}
+    >
+      <Paper {...props} />
+    </Draggable>
+  );
+}
+
+interface IProps {
+  open: boolean;
+  boundary: IBoundaryInfo;
+  getPlateHue: (plateId?: number) => number | undefined;
+  onAssign: (type: BoundaryType) => void;
+  onClose: () => void;
+}
+// patterned after https://mui.com/components/dialogs/#draggable-dialog
+const BoundaryConfigDialog = observer(({ open, boundary, getPlateHue, onAssign, onClose }: IProps) => {
+  return (
+    <Dialog
+      className={css.boundaryConfigDialog}
+      open={open}
+      onClose={onClose}
+      PaperComponent={PaperComponent}
+      aria-labelledby="draggable-dialog-title"
+    >
+      <DialogTitle style={{ cursor: "move" }} id="draggable-dialog-title">
+        Plate Boundary Type
+        <CloseIcon onClick={onClose} />
+      </DialogTitle>
+      <div className={css.dividerLine} />
+      <DialogActions>
+        <BoundaryOption boundary={boundary} type="convergent" getPlateHue={getPlateHue} onAssign={onAssign}/>
+        <BoundaryOption boundary={boundary} type="divergent" getPlateHue={getPlateHue} onAssign={onAssign}/>
+      </DialogActions>
+    </Dialog>
+  );
+});
+export default BoundaryConfigDialog;
+
+// The spec contains a mapping from plate color to arrow color.
+// The code determines the plate color from the plate's hue value and its elevation.
+// For this purpose we statically map the spec's plate colors to hue using https://htmlcolors.com/hex-to-hsl.
+// Then in code we find the entry with the hue closest to the plate's hue.
+// The entry index is then turned into a color class so that the actual arrow colors are in the .less file.
+const plateHues = [322, 126, 30, 263, 199, 34, 162, 69, 210, 301];
+
+export function getArrowColorClassFromPlateHue(plateHue: number) {
+  let hueIndex = 0;
+  let minHueDistance = 180;
+  plateHues.forEach((hue, index) => {
+    let hueDistance = Math.abs(hue - plateHue);
+    (hueDistance > 180) && (hueDistance -= 180);
+    if (hueDistance < minHueDistance) {
+      hueIndex = index;
+      minHueDistance = hueDistance;
+    }
+  });
+  return `arrow-color-${hueIndex}`;
+}
```

Comment:
The spec uses custom colors designed by Michael that correspond to each plate color, apparently to make the colors more vibrant. The question of whether we should use the actual plate color rather than the custom colors from the spec is a question for the design team. At least with this implementation if we ever encounter a plate hue for which there isn't a custom color in the spec (and hence in the code) we'll still pick the closest color.